source/Coop/Mod/Data/GameData.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Common;
+using Coop.Mod.Serializers.Custom;
+using NLog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using TaleWorlds.CampaignSystem;
+
+namespace Coop.Mod.Data
+{
+    [Serializable]
+    public class GameData
```

Comment:
I'm not in love with the idea of sending the entire state of the current game via a big blob. I think there are optimizations to be had here where we send pages of objects/load things periodically as need be. The server should manage everything, I think the clients should only worry about things that are relevant to them.